jujugui/static/gui/src/app/components/status/status.js

```diff
@@ -109,6 +109,59 @@ class Status extends React.Component {
   }
 
   /**
+    Navigate to the chosen application.
+    @param appId {String} The id of the application to display.
+  */
+  _navigateToApplication(appId) {
+    this.props.changeState({
+      gui: {
+        inspector: {
+          id: appId
+        }
+      }
+    });
+  }
+
+  /**
+    Navigate to the chosen charm.
+    @param charmID {String} The id of the charm to display.
+  */
+  _navigateToCharm(appId) {
+    this.props.changeState({store: appId});
+  }
+
+  /**
+    Navigate to the chosen unit.
+    @param unitId {String} The id of the unit to display in the format
+      'service-id/unit-id'.
```

Comment:
Hyphens are ok, unit-id can be confusing as I expect it to be mysql/0, not just 0. That's why I suggested (app name/unit number).